### ISSUE TITLE
I.. Thought we were done with this..

### ISSUE THREAD

**Issue by MInerJake26** [Ellpeck/NaturesAura#105]
---- Minecraft Crash Report ----
// There are four lights!

Time: 4/9/20 3:34 PM
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
	at vazkii.patchouli.client.handler.BookRightClickHandler.getHoveredEntry(BookRightClickHandler.java:113) ~[?:1.15.2-1.2-30] {re:classloading}
	at vazkii.patchouli.client.handler.BookRightClickHandler.onRenderHUD(BookRightClickHandler.java:48) ~[?:1.15.2-1.2-30] {re:classloading}
	at net.minecraftforge.eventbus.ASMEventHandler_179_BookRightClickHandler_onRenderHUD_Post.invoke(.dynamic) ~[?:?] {}
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:80) ~[eventbus-2.0.0-milestone.1-service.jar:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258) ~[eventbus-2.0.0-milestone.1-service.jar:?] {}
	at net.minecraftforge.client.gui.ForgeIngameGui.post(ForgeIngameGui.java:830) ~[?:?] {re:classloading}
	at net.minecraftforge.client.gui.ForgeIngameGui.func_175180_a(ForgeIngameGui.java:221) ~[?:?] {re:classloading}
	at net.minecraft.client.renderer.GameRenderer.func_195458_a(GameRenderer.java:460) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:925) [?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:554) [?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(SourceFile:204) [?:?] {re:classloading}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51] {}
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:56) [forge-1.15.2-31.1.27.jar:31.1] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider$$Lambda$471/826690115.call(Unknown Source) [forge-1.15.2-31.1.27.jar:31.1] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-5.0.0-milestone.4.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-5.0.0-milestone.4.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-5.0.0-milestone.4.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:81) [modlauncher-5.0.0-milestone.4.jar:?] {re:classloading}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:65) [modlauncher-5.0.0-milestone.4.jar:?] {re:classloading}

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at vazkii.patchouli.client.handler.BookRightClickHandler.getHoveredEntry(BookRightClickHandler.java:113)
	at vazkii.patchouli.client.handler.BookRightClickHandler.onRenderHUD(BookRightClickHandler.java:48)
	at net.minecraftforge.eventbus.ASMEventHandler_179_BookRightClickHandler_onRenderHUD_Post.invoke(.dynamic)
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:80)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258)
	at net.minecraftforge.client.gui.ForgeIngameGui.post(ForgeIngameGui.java:830)
	at net.minecraftforge.client.gui.ForgeIngameGui.func_175180_a(ForgeIngameGui.java:221)

-- Affected level --
Details:
	All players: 1 total; [ClientPlayerEntity['MinerJake26'/141, l='MpServer', x=-167.42, y=69.00, z=-305.66]]
	Chunk stats: Client Chunk Cache: 2025, 1537
	Level dimension: DimensionType{minecraft:overworld}
	Level name: MpServer
	Level seed: 2399773522954201855
	Level generator: ID 07 - biomesoplenty, ver 0. Features enabled: false
	Level generator options: {}
	Level spawn location: World: (-21,63,15), Chunk: (at 11,3,15 in -2,0; contains blocks -32,0,0 to -17,255,15), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Level time: 659671 game time, 132720 day time
	Known server brands: 
	Level was modded: false
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: true), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Server brand: forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.world.ClientWorld.func_72914_a(ClientWorld.java:442)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:1764)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:573)
	at net.minecraft.client.main.Main.main(SourceFile:204)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:56)
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider$$Lambda$471/826690115.call(Unknown Source)
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72)
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:81)
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:65)

-- System Details --
Details:
	Minecraft Version: 1.15.2
	Minecraft Version ID: 1.15.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 2121051776 bytes (2022 MB) / 4294967296 bytes (4096 MB) up to 4294967296 bytes (4096 MB)
	CPUs: 6
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx4G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	ModLauncher: 5.0.0-milestone.4+67+b1a340b
	ModLauncher launch target: fmlclient
	ModLauncher naming: srg
	ModLauncher services: 
		/eventbus-2.0.0-milestone.1-service.jar eventbus PLUGINSERVICE 
		/forge-1.15.2-31.1.27.jar object_holder_definalize PLUGINSERVICE 
		/forge-1.15.2-31.1.27.jar runtime_enum_extender PLUGINSERVICE 
		/accesstransformers-2.0.4-shadowed.jar accesstransformer PLUGINSERVICE 
		/forge-1.15.2-31.1.27.jar capability_inject_definalize PLUGINSERVICE 
		/forge-1.15.2-31.1.27.jar runtimedistcleaner PLUGINSERVICE 
		/forge-1.15.2-31.1.27.jar fml TRANSFORMATIONSERVICE 
		/MixinBootstrap-1.0.2.jar mixinbootstrap TRANSFORMATIONSERVICE 
		/MixinBootstrap-1.0.2.jar mixin PLUGINSERVICE 
		/MixinBootstrap-1.0.2.jar mixinbootstrap PLUGINSERVICE 
	FML: 31.1
	Forge: net.minecraftforge:31.1.27
	FML Language Providers: 
		javafml@31.1
		minecraft@1
	Mod List: 
		forge-1.15.2-31.1.27-client.jar Minecraft {minecraft@1.15.2 DONE}
		mcw-bridges-1.0.4fix-mc1.15.2.jar Macaw's Bridges {mcwbridges@1.0.4 DONE}
		useful_backpacks-1.15.2-1.9.4.63.jar Useful Backpacks {usefulbackpacks@1.9.4.63 DONE}
		inventorysorter-1.15.2-17.0.0.jar Simple Inventory Sorter {inventorysorter@17.0.0 DONE}
		AmbientSounds_v3.0.19_mc1.15.2.jar Ambient Sounds {ambientsounds@3.0.3 DONE}
		MouseTweaks-2.13-mc1.15.1.jar Mouse Tweaks {mousetweaks@2.13 DONE}
		Cucumber-1.15.2-3.0.3.jar Cucumber Library {cucumber@3.0.3 DONE}
		NetherPortalFix_1.15.2-6.0.0.jar NetherPortalFix {netherportalfix@6.0.0 DONE}
		SilentLib-1.15.2-4.6.0+48.jar Silent Lib {silentlib@4.6.0+48 DONE}
		BiomesOPlenty-1.15.2-10.0.0.345-universal.jar Biomes O' Plenty {biomesoplenty@1.15.2-10.0.0.342 DONE}
		mining-helmet-1.15.2-1.0.7.jar Mining Helmet {mining_helmet@1.0.7 DONE}
		Blur-1.1.0-1.jar Blur {blur@1.1.0-1 DONE}
		valhelsia_structures-15.0.2a.jar Valhelsia Structures {valhelsia_structures@15.0.2a DONE}
		SilentGear-1.15.2-1.5.4+79.jar Silent Gear {silentgear@1.5.4+79 DONE}
		jei-1.15.2-6.0.0.2.jar Just Enough Items {jei@6.0.0.2 DONE}
		simplefarming-1.15.2-1.2.7.jar Simple Farming {simplefarming@1.15.2-1.2.7 DONE}
		DoggyTalents-1.15.2-1.15.2.3.jar Doggy Talents {doggytalents@1.15.2.3 DONE}
		goblintraders-1.2.0-1.15.2.jar Goblin Traders {goblintraders@1.2.0 DONE}
		SilentMechanisms-1.15.2-0.7.0+50.jar Silent's Mechanisms {silents_mechanisms@0.7.0+50 DONE}
		Patchouli-1.15.2-1.2-30.jar Patchouli {patchouli@1.15.2-1.2-30 DONE}
		gravestone-1.17.3.jar Gravestone Mod {gravestone@1.17.3 DONE}
		journeymap-1.15.2-5.7.0beta1.jar Journeymap {journeymap@1.15.2-5.7.0beta1 DONE}
		OreExcavation-1.7.151.jar Ore Excavation {oreexcavation@NONE DONE}
		NaturesCompass-1.15.2-1.8.5.jar Nature's Compass {naturescompass@1.15.2-1.8.5 DONE}
		FruitTrees-1.15.2-1.1.6.jar Fruit Trees {fruittrees@1.1.6 DONE}
		SereneSeasons-1.15.2-3.0.0.68-universal.jar Serene Seasons {sereneseasons@1.15.2-3.0.0.68 DONE}
		performant-1.15-1.7.jar Performant {performant@1.7 DONE}
		VanillaTweaks-1.15.2.1.5.18.jar VanillaTweaks {vanillatweaks@1.15.2.1.5.18 DONE}
		obfuscate-0.3.4-1.15.2.jar Obfuscate {obfuscate@0.3.4 DONE}
		DefaultSettings-1.15.2-2.1.5.jar DefaultSettings {defaultsettings@2.1.5 DONE}
		snowundertrees-1.15.1-v1.0.jar Snow Under Trees {snowundertrees@v1.0 DONE}
		u_team_core-1.15.2-2.11.0.156.jar U Team Core {uteamcore@2.11.0.156 DONE}
		PickleTweaks-1.15.2-4.0.1.jar Pickle Tweaks {pickletweaks@4.0.1 DONE}
		NaturesAura-23.5.jar Nature's Aura {naturesaura@version DONE}
		nhacampfire_1.15.2-1.5.jar No Hostiles Around Campfire {nhacampfire@1.5 DONE}
		Waddles-1.15.2-0.8.5.jar Waddles {waddles@1.15.2-0.8.5 DONE}
		furniture-7.0.0-pre16-1.15.1.jar MrCrayfish's Furniture Mod {cfm@7.0.0-pre16 DONE}
		AppleSkin-mc1.15.2-forge-1.0.13.jar AppleSkin {appleskin@1.0.13 DONE}
		SnowRealMagic-1.15.2-1.7.2.jar Snow! Real Magic! {snowrealmagic@1.7.2 DONE}
		Kiwi-1.15.2-2.6.5.jar Kiwi {kiwi@2.6.5 DONE}
		the_bumblezone-1.15.2-1.0.4.jar The Bumblezone {the_bumblezone@1.15.2-1.0.4 DONE}
		forge-1.15.2-31.1.27-universal.jar Forge {forge@31.1.27 DONE}
		Aquaculture-1.15.2-2.0.11.jar Aquaculture 2 {aquaculture@1.15.2-2.0.11 DONE}
		SolarGeneration-1.15.2-1.2.0.jar Solar Generation {solargeneration@1.2.0 DONE}
		SoundFilters-0.13_for_1.15.2.jar Sound Filters {soundfilters@0.13_for_1.15.2 DONE}
		refinedstorage-1.8.1.jar Refined Storage {refinedstorage@1.8.1 DONE}
		refinedstorageaddons-0.6.1.jar Refined Storage Addons {refinedstorageaddons@0.6.1 DONE}
		MysticalAgriculture-1.15.2-3.0.1.jar Mystical Agriculture {mysticalagriculture@3.0.1 DONE}
	Kiwi Modules: 
		fruittrees:fruittrees
		fruittrees:hybridization
		kiwi:contributors
		snowrealmagic:snowrealmagic
	Launched Version: 1.15.2-forge-31.1.27
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: GeForce GTX 1050 Ti/PCIe/SSE2 GL version 4.6.0 NVIDIA 436.50, NVIDIA Corporation
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'forge'
	Type: Client (map_client.txt)
	Resource Packs: vanilla
	Current Language: English (US)
	CPU: 6x Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz

**Comment by Ellpeck** [comment]
Please try to find out which mod exactly is causing this issue to occur.

**Comment by MInerJake26** [comment]
I updated patchouli first and it didn’t work then I updated Nature’s Aura and it did, until I crashed again. The next two logs crashed me, and then it miraculously worked. I haven’t had the problem again yet. The first time it happened, it was because I opened a barrel with the book. Then it was just because I had the entity. I think when I tried to log in all it remember was the buggy entity or something. Idk I don’t know much about how mods are programmed.
